hoomd/hpmc/CMakeLists.txt

```
@@ -73,38 +97,67 @@ set(_hpmc_headers
     )
 
 set(_hpmc_cu_sources IntegratorHPMCMonoGPU.cu
-                     all_kernels_sphere.cu
-                     all_kernels_convex_polygon.cu
-                     all_kernels_simple_polygon.cu
-                     all_kernels_spheropolygon.cu
-                     all_kernels_polyhedron.cu
-                     all_kernels_ellipsoid.cu
-                     all_kernels_faceted_ellipsoid.cu
-                     all_kernels_sphinx.cu
-                     all_kernels_union_spheropolyhedron.cu
-                     all_kernels_union_faceted_ellipsoid.cu
-                     all_kernels_union_sphere.cu
-                     all_kernels_convex_polyhedron.cu
-                     all_kernels_convex_spheropolyhedron.cu
+                     IntegratorHPMCMonoGPUDepletants.cu
+                     UpdaterClustersGPU.cu
                      )
 
+set(_hpmc_kernel_templates kernel_free_volume
+                           kernel_gen_moves
+                           kernel_narrow_phase
+                           kernel_insert_depletants
+                           kernel_update_pdata
+                           kernel_cluster_overlaps
+                           kernel_cluster_depletants
+                           kernel_cluster_transform
+                           kernel_depletants_auxilliary_phase1
+                           kernel_depletants_auxilliary_phase2)
+
+if(ENABLE_HIP)
+    # expand the shape x GPU kernel matrix of template instantiations
+    foreach(KERNEL ${_hpmc_kernel_templates})
+        foreach(SHAPE ${_hpmc_gpu_shapes})
+            set(SHAPE_INCLUDE ${SHAPE}.h)
+            set(IS_UNION_SHAPE FALSE)
+            set(_kernel_cu ${KERNEL}_${SHAPE}.cu)
+            configure_file(${KERNEL}.cu.in ${_kernel_cu} @ONLY)
+            set(_hpmc_cu_sources ${_hpmc_cu_sources} ${_kernel_cu})
+        endforeach()
+
+        foreach(SHAPE ${_hpmc_gpu_union_shapes})
+            set(SHAPE_INCLUDE ${SHAPE}.h)
+            set(_kernel_cu ${KERNEL}_union_${SHAPE}.cu)
+            set(IS_UNION_SHAPE TRUE)
+            configure_file(${KERNEL}.cu.in ${_kernel_cu} @ONLY)
+            set(_hpmc_cu_sources ${_hpmc_cu_sources} ${_kernel_cu})
+        endforeach()
+    endforeach()
+endif(ENABLE_HIP)
+
 if (ENABLE_HIP)
 set(_cuda_sources ${_hpmc_cu_sources})
 endif (ENABLE_HIP)
 
 pybind11_add_module(_hpmc SHARED ${_hpmc_sources} ${_cuda_sources} ${_hpmc_headers} NO_EXTRAS)
 # alias into the HOOMD namespace so that plugins and symlinked components both work
 add_library(HOOMD::_hpmc ALIAS _hpmc)
-
 if (APPLE)
 set_target_properties(_hpmc PROPERTIES INSTALL_RPATH "@loader_path/..;@loader_path")
 else()
 set_target_properties(_hpmc PROPERTIES INSTALL_RPATH "\$ORIGIN/..;\$ORIGIN")
 endif()
 
+if(ENABLE_HIP)
+    # enable out-of-source build
+    target_include_directories(_hpmc PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
```

The source directory may not be available once hoomd is installed, or the source directory may have headers that are not compatible with the installation.  This should be `PRIVATE` so that plugins don't attempt to include files from the source directory.
```suggestion
    target_include_directories(_hpmc PRIVATE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
```

---

done (sorry, I could have just committed your suggestion).